examples/text-classification/run_xnli.py

```diff
@@ -16,611 +16,315 @@
 """ Finetuning multi-lingual models on XNLI (e.g. Bert, DistilBERT, XLM).
     Adapted from `examples/text-classification/run_glue.py`"""
 
-
-import argparse
-import glob
 import logging
 import os
 import random
+import sys
+from dataclasses import dataclass, field
+from typing import Optional
 
 import numpy as np
 import torch
-from torch.utils.data import DataLoader, RandomSampler, SequentialSampler, TensorDataset
-from torch.utils.data.distributed import DistributedSampler
-from tqdm import tqdm, trange
+from datasets import load_dataset, load_metric
 
 import transformers
 from transformers import (
-    WEIGHTS_NAME,
-    AdamW,
     AutoConfig,
     AutoModelForSequenceClassification,
     AutoTokenizer,
-    get_linear_schedule_with_warmup,
+    DataCollatorWithPadding,
+    EvalPrediction,
+    HfArgumentParser,
+    Trainer,
+    TrainingArguments,
+    default_data_collator,
+    set_seed,
 )
-from transformers import glue_convert_examples_to_features as convert_examples_to_features
-from transformers import xnli_compute_metrics as compute_metrics
-from transformers import xnli_output_modes as output_modes
-from transformers import xnli_processors as processors
-from transformers.trainer_utils import is_main_process
-
+from transformers.trainer_utils import get_last_checkpoint, is_main_process
 
-try:
-    from torch.utils.tensorboard import SummaryWriter
-except ImportError:
-    from tensorboardX import SummaryWriter
 
+task_to_keys = {"xnli": ("premise", "hypothesis")}
```

Comment:
I think we can just use `"premise"` and `"hypothesis"` later on, since there is only one task we don't need this variable anymore.

Reply:
I've removed dict `task_to_keys` as we can later refer directly to "xnli", "premise", and "hypothesis" 
https://github.com/huggingface/transformers/blob/0f6858f34ff5b56a451f893548269904f3efb751/examples/text-classification/run_xnli.py#L237-L238
https://github.com/huggingface/transformers/blob/0f6858f34ff5b56a451f893548269904f3efb751/examples/text-classification/run_xnli.py#L204

https://github.com/huggingface/transformers/blob/0f6858f34ff5b56a451f893548269904f3efb751/examples/text-classification/run_xnli.py#L314
https://github.com/huggingface/transformers/blob/0f6858f34ff5b56a451f893548269904f3efb751/examples/text-classification/run_xnli.py#L318
https://github.com/huggingface/transformers/blob/0f6858f34ff5b56a451f893548269904f3efb751/examples/text-classification/run_xnli.py#L256
https://github.com/huggingface/transformers/blob/0f6858f34ff5b56a451f893548269904f3efb751/examples/text-classification/run_xnli.py#L186-L188
https://github.com/huggingface/transformers/blob/0f6858f34ff5b56a451f893548269904f3efb751/examples/text-classification/run_xnli.py#L190